myhpom/tests/test_signup_view.py

```diff
@@ -0,0 +1,60 @@
+from django.core.urlresolvers import reverse
+from django.test import Client, TestCase
+from myhpom.models import User, UserDetails, State
+
+
+class SignupTestCase(TestCase):
+    """
+    * GET retries page
+    * POST invalid data returns to the page
+    * POST valid data in non-supported state for non-existing user redirects to 'next_steps'
+    * POST valid data in supported state for non-existing user redirects to 'choose_network'
+    """
+
+    def setUp(self):
+        self.client = Client()
+        self.url = reverse('myhpom:signup')
+        self.form_data = {
+            'first_name': 'A',
+            'last_name': 'B',
+            'email': 'ab@example.com',
+            'password': 'Abbbbbbb1@',
+            'password_confirm': 'Abbbbbbb1@',
+            'state': 'NC',  # a supported state
+            'accept_tos': True,
+        }
+
+    def test_get_signup(self):
+        """GET the signup page should work"""
+        response = self.client.get(self.url)
+        self.assertEqual(200, response.status_code)
+        self.assertTemplateUsed('myhpom/accounts/signup.html')
+
+    def test_post_signup_invalid(self):
+        """invalid signup should redisplay signup page"""
+        data = {}  # empty signup data is certainly invalid!
+        response = self.client.post(self.url, data=data)
+        self.assertEqual(200, response.status_code)
+        self.assertTemplateUsed('myhpom/accounts/signup.html')
+
+    def test_post_signup_valid_supported(self):
+        """valid signup with a supported state should redirect to choose_network"""
+        data = self.form_data
+        data['state'] = State.objects.filter(supported=True)[0].name
+        response = self.client.post(self.url, data=data)
+        # reset database state
+        User.objects.get(email=self.form_data['email']).delete()
```

Comment:
These `User.objects...delete()` isn't needed since you're using django's TestCase.

Reply:
(same for the delete() in the test case below)

Reply:
Doing it this way enables us to keepdb / with migrations, right? We could move it to tearDown, but it seems necessary to include it if we're going to test via `REUSE_DB=1 python manage.py test --nomigrations --keepdb`

Reply:
Right, --keepdb avoids the overhead of the initial creation/migration of the DB, by keeping the test database around for reuse after the full test run completes. Between every test case the action of the previous test case are undone (usually by rolling back the transaction the test case was run in).  --keepdb doesn't affect what happens between test cases in a single test run.

Reply:
Got it, thanks.